### PR TITLE
[DLS] Separate access control syncs task pool from content syncs task pool

### PR DESCRIPTION
**Pull request by timgrein**
## Closes https://github.com/elastic/enterprise-search-team/issues/4583

This PR introduces a separate task pool for access control syncs, so content and access control syncs don't interfer too much with each other. Both can be configured and therefore scaled independently from each other.

It also introduces two new configuration entries:

- `max_concurrent_content_syncs` (`max_concurrent_syncs` still works, but the new setting should be used as it's more explicit)
- `max_concurrent_access_control_syncs`

#### Pre-Review Checklist
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] if there is no GH issue, please create it. Each PR should have a link to an issue
- [x] this PR has a thorough description
- [x] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)
- [x] Contributed any configuration settings changes to the configuration reference

## Release Note

Deprecated `max_concurrent_syncs` entry in `config.yml`, which is replaced by `max_concurrent_content_syncs` specifying the maximum number of concurrent content syncs run by a connector service instance.
Added `max_concurrent_access_control_syncs` to `config.yml` specifying the maximum number of concurrent access control syncs run by a connector service instance.

### CODE REVIEW

File: docs/CONFIG.md [hunk 1]
```diff
@@ -33,7 +33,9 @@ Configuration lives in [config.yml](../config.yml).
   - `preflight_idle`: The number of seconds to wait between each pre-flight check. Defaults to 30.
   - `max_errors`: The maximum number of errors allowed in one event loop.
   - `max_errors_span`: The number of seconds to reset `max_errors` count.
-  - `max_concurrent_syncs`: The maximum number of concurrent syncs. Defaults to 1.
+  - `max_concurrent_syncs`: The maximum number of concurrent content syncs. Defaults to 1. (deprecated)
```
Comment (seanstory) [inline docs/CONFIG.md:36]:
```suggestion
  - `max_concurrent_syncs`: (Deprecated. Use `max_concurrent_content_syncs`) The maximum number of concurrent content syncs. Defaults to 1. 
```

File: tests/services/test_job_execution.py [hunk 2]
```diff
@@ -279,4 +368,59 @@ async def test_access_control_job_execution_with_dls_feature_flag_disabled(
 
     await create_and_run_service()
 
-    concurrent_tasks_mock.put.assert_awaited()
+    content_syncs_tasks_mock.put.assert_not_awaited()
+    access_control_syncs_tasks_mock.put.assert_not_awaited()
+
+
+@pytest.mark.asyncio
+async def test_job_execution_with_unknown_job_type(
```
Comment (seanstory) [inline tests/services/test_job_execution.py:376]: Can we also assert the relevant error is logged?

**Review by seanstory — COMMENTED** [review]
Looking good. Few small nits below.

Can you also add the `release_note` label and a brief "Release Note" section to the PR description noting the deprecated config and its replacement?

File: config.yml [hunk 3]
```diff
@@ -23,6 +23,8 @@ service:
   max_errors: 20
   max_errors_span: 600
   max_concurrent_syncs: 1
```
[the remark this answers is not in the document]
Comment (timgrein) [reply config.yml:25]: Yes, that'll be a fallback. Sounds good, removing it 👍

File: connectors/services/job_execution.py [hunk 4]
```diff
@@ -18,7 +18,25 @@
...
+
+
+def load_max_concurrent_content_syncs(config):
+    max_concurrent_content_syncs = config.get("max_concurrent_content_syncs")
+
+    if max_concurrent_content_syncs is not None:
+        return max_concurrent_content_syncs
+
+    # keep for backwards compatibility
+    return config.get("max_concurrent_syncs", DEFAULT_MAX_CONCURRENT_CONTENT_SYNCS)
```
[the remark this answers is not in the document]
Comment (timgrein) [reply connectors/services/job_execution.py:41]: Good point, added it 👍

File: config.yml [hunk 5]
```diff
@@ -23,6 +23,8 @@ service:
...
+  max_concurrent_content_syncs: 1
+  max_concurrent_access_control_syncs: 1
```
[the remark this answers is not in the document]
Comment (timgrein) [reply config.yml:26]: We already have an issue for restructuring the whole `service` configuration section. I'll add that as a possible requirement to this issue.

File: tests/services/test_job_execution.py (hunk 2 above)
Comment (timgrein) [reply tests/services/test_job_execution.py:376]: We've decided to stay away from using assertions on log messages for several reasons (more fragile tests, focus should be on the functionality, in this case "not executing a job" etc.). So, I'm hesitant to reintroduce it here.

WDYT?

**Review by artem-shelkovnikov — APPROVED** [review]

File: connectors/services/job_execution.py [hunk 6]
```diff
@@ -27,31 +49,77 @@ class JobExecutionService(BaseService):
     def __init__(self, config):
         super().__init__(config)
         self.idling = self.service_config["idling"]
-        self.concurrent_syncs = self.service_config.get(
-            "max_concurrent_syncs", DEFAULT_MAX_CONCURRENT_SYNCS
+        self.max_concurrent_content_syncs = load_max_concurrent_content_syncs(
+            self.service_config
+        )
+        self.max_concurrent_access_control_syncs = (
```
Comment (wangch079) [inline connectors/services/job_execution.py:55]: Just curious, why the outer parenthesis? added automatically by `make autoformat`?

**Review by wangch079 — COMMENTED** [review]
Don't forget to update https://github.com/elastic/ent-search/blob/main/dist/shared/cloud/coservices/python-connectors.yml

**Pull request merged**